dll/win32/setupapi/queue.c

```diff
@@ -1092,25 +1088,45 @@ static BOOL do_file_copyW( LPCWSTR source, LPCWSTR target, DWORD style,
         return FALSE;
     }
 
+    if (!GetTempFileNameW(TempPath, L"", 0, TempFile))
+    {
+        ERR("GetTempFileNameW(%s) error\n", debugstr_w(TempPath));
+
+        /* Close the source handle */
+        LZClose(hSource);
+
+        return FALSE;
+    }
+
     /* Extract the compressed file to a temp location */
     hTemp = LZOpenFileW(TempFile, &OfStruct, OF_CREATE);
     if (hTemp < 0)
     {
-        DWORD dwLastError = GetLastError();
-
         ERR("LZOpenFileW(2) error %d %s\n", (int)hTemp, debugstr_w(TempFile));
 
         /* Close the source handle */
         LZClose(hSource);
 
-        /* Restore error condition triggered by LZOpenFileW */
```

Comment:
I'm not sure I understand why this was removed. I'm assuming we'd still want the last error to be sensible?

Reply:
I'd argue this should be put back, and also added to the GetTempFileNameW failure handler and the 'if (lRes < 0)' handler at line 1121
Do we have any tests that confirm?

Reply:
@gedmurphy Your comment sounds like a blocker for merging this PR to master. If that's true, please signal this by clicking Review changes -> Request changes. Otherwise, according to the new PR guidelines, this PR fulfills all requirements and would be merged at the next occasion.

Of course, @ThFabba is also free to simply address the comment and fix the code accordingly.